master/buildbot/process/builder.py

```diff
@@ -261,10 +263,21 @@ def getAvailableWorkers(self):
         return [wfb for wfb in self.workers if wfb.isAvailable()]
     deprecatedWorkerClassMethod(locals(), getAvailableWorkers)
 
-    def canStartWithWorkerForBuilder(self, workerforbuilder):
+    @defer.inlineCallbacks
+    def canStartWithWorkerForBuilder(self, workerforbuilder, buildrequests):
```

Comment:
If we want absolutely preserve this API as is, buildrequests could be set to None by default and the method would fail if any of the locks were renderable when the method was called without build request information.